client/src/main/java/io/pravega/client/tables/KeyValueTable.java

```diff
@@ -98,197 +122,147 @@
+     * matching the given {@code entry}'s {@link TableEntry#getKey()}).
+     * <li> Conditional Update if the {@code entry}'s {@link TableEntry#getKey()} {@link TableKey#getVersion()} does not
+     * equals {@link Version#NOT_EXISTS} or {@link Version#NO_VERSION} (i.e., if the {@link TableEntry} was created using
+     * {@link TableEntry#versioned}). The {@link TableEntry} will only be inserted if there exists a {@link TableKey} in
+     * the {@link KeyValueTable} matching the given {@code entry}'s {@link TableEntry#getKey()} and both key versions match.
+     * </ul>
+     * <p>
+     * All other invocations will be executed as Unconditional Updates or Insertions ({@code entry.getKey().getVersion()} equals
+     * {@link Version#NO_VERSION} - such as when the entry was created using {@link TableEntry#unversioned}).
      *
-     * @param keyFamily (Optional) The Key Family for the Entry. If null, this Entry will not be associated with any
-     *                  Key Family.
-     * @param key       The Key to insert or update.
-     * @param value     The Value to be associated with the Key.
+     * @param entry The {@link TableEntry} to insert or update.
      * @return A CompletableFuture that, when completed, will contain the {@link Version} associated with the newly
      * inserted or updated entry.
      */
-    CompletableFuture<Version> put(@Nullable String keyFamily, @NonNull KeyT key, @NonNull ValueT value);
+    CompletableFuture<Version> put(@NonNull TableEntry entry);
 
     /**
-     * Conditionally inserts a new Entry in the {@link KeyValueTable} if the given Key is not already present.
-     *
-     * @param keyFamily (Optional) The Key Family for the Entry. If null, this Entry will not be associated with any
-     *                  Key Family.
-     * @param key       The Key to insert.
-     * @param value     The Value to be associated with the Key.
-     * @return A CompletableFuture that, when completed, will contain the {@link Version} associated with the newly
-     * inserted or updated entry. Notable exceptions:
-     * <ul>
-     * <li>{@link ConditionalTableUpdateException} If the Key is already present in the {@link KeyValueTable} for the
-     * provided Key Family.
-     * See the {@link KeyValueTable} doc for more details on Conditional Update Responses.
-     * </ul>
-     */
-    CompletableFuture<Version> putIfAbsent(@Nullable String keyFamily, @NonNull KeyT key, @NonNull ValueT value);
-
-    /**
-     * Unconditionally inserts new or updates existing {@link TableEntry} instances that belong to the same Key Family
+     * Inserts new or updates existing {@link TableEntry} instances that have the same {@link TableKey#getPrimaryKey()}
      * into this {@link KeyValueTable}. All changes are performed atomically (either all or none will be accepted).
+     * <p>
+     * See {@link #put(TableEntry)} for details on Conditional/Unconditional Updates or Insertions. Conditional and
```

Comment:
Ok, so to confirm, even if there is only a single conditional update in a set of many updates and the condition fails, the whole batch fails, including the unconditional ones, yes?

Reply:
Yes. the whole thing is rejected with no side effects.

If this text is not clear, I'm open to suggestions how to rephrase it:
```
and will prevent the committal of the entire batch of updates if it fails to meet the condition.
```

Reply:
I'd say `... prevent the entire batch from committing`. I think. `committal` means something else.

Reply:
Rephrased.